### PR TITLE
[Fix] Bunch of planetoids fixes

### PR DESCRIPTION
**Pull request by hyperioo**
## About The Pull Request

After the release of planetary exploration, we got lots of feedback about bugs or oversights.

## Why It's Good For The Game

Improve planetary exploration experience.

## Changelog
:cl: Hyperio
add: You can now place lattice and floors on planets
fix: Exploration flags can now be put down on planets
fix: Roombas no longer carry wood floor overlay with them
fix: Mineral overlays should no longer appear on ruins turfs
fix: Foutain no longer duplicates organs if you have prosthetics
fix: Lava no longer melts reality (ghosts, lights)
fix: Officer uniform and naked corpses in ruins
fix: Blacklist ruins exclusive items from spawning on Eris
tweak: All access removed from the crashed pod corpses' IDs 
/:cl:

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -762,5 +762,10 @@ its easier to just keep the beam vertical.
 /atom/proc/lava_act()
 	visible_message("<span class='danger'>\The [src] sizzles and melts away, consumed by the lava!</span>")
 	playsound(src, 'sound/effects/flare.ogg', 100, 3)
-	qdel(src)
+	if(ismob(src))
+		var/mob/M = src
+		M.death(FALSE, FALSE)
+		qdel(src)
+	else
+		qdel(src)
```
Comment (AlexMorgan3817) [inline code/game/atoms.dm:770]:
```suggestion
	qdel(src)
```

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -104,7 +104,7 @@
 	var/obj/item/stack/flag/F = locate() in get_turf(src)
 
 	var/turf/T = get_turf(src)
-	if(!T || !istype(T,/turf/simulated/floor/asteroid))
+	if(!T || !(istype(T,/turf/simulated/floor/asteroid) || istype(T, /turf/simulated/floor/exoplanet)))
```
Comment (AlexMorgan3817) [inline code/modules/mining/mine_items.dm:107]: Id you check istype it automaticaly check isnull

```suggestion
	if(!istype(T,/turf/simulated/floor/asteroid) || istype(T, /turf/simulated/floor/exoplanet))
```

File: code/modules/overmap/exoplanets/helpers.dm [hunk 3]
```diff
@@ -116,7 +116,7 @@ GLOBAL_LIST_EMPTY(banned_ruin_ids)
 	template.load(central_turf,centered = TRUE)
 	var/datum/map_template/ruin = template
 	if(istype(ruin))
-		new /obj/effect/landmark/ruin(central_turf, ruin)
+		new /obj/effect/landmark/ruin/automatic/clearing(central_turf, ruin, round(sqrt((ruin.width * 0.5)**2 + (ruin.height)**2)))
```
Comment (AlexMorgan3817) [inline code/modules/overmap/exoplanets/helpers.dm:119]:
```suggestion
		new /obj/effect/landmark/ruin/automatic/clearing(central_turf, ruin, round(sqrt((ruin.width / 2) ** 2 + (ruin.height) ** 2)))
```

File: code/modules/overmap/exoplanets/helpers.dm [hunk 4]
```diff
@@ -145,4 +145,21 @@ GLOBAL_LIST_EMPTY(banned_ruin_ids)
...
-	. = ..()
\ No newline at end of file
+	. = ..()
+
+//Subtype that calls explosion on init to clear space for shuttles
+/obj/effect/landmark/ruin/automatic/clearing
+	var/radius = 0
+
+/obj/effect/landmark/ruin/automatic/clearing/New(loc, my_ruin_template, ruin_radius)
+	..(loc, my_ruin_template)
```
Comment (AlexMorgan3817) [inline code/modules/overmap/exoplanets/helpers.dm:155]:
```suggestion
	. = ..(loc, my_ruin_template)
```

**Review by AlexMorgan3817 — CHANGES_REQUESTED** [review]

File: code/modules/mining/mine_items.dm (hunk 2 above)
Comment (hyperioo) [reply code/modules/mining/mine_items.dm:107]: Are you sure the ! will apply to (istype() || istype()) and not only to the first istype() ?

Comment (AlexMorgan3817) [reply code/modules/mining/mine_items.dm:107]: Ah, forgot it, sorry

File: code/modules/overmap/exoplanets/helpers.dm [hunk 5]
```diff
@@ -116,7 +116,7 @@ GLOBAL_LIST_EMPTY(banned_ruin_ids)
 	template.load(central_turf,centered = TRUE)
 	var/datum/map_template/ruin = template
 	if(istype(ruin))
-		new /obj/effect/landmark/ruin/automatic/clearing(central_turf, ruin, round(sqrt((ruin.width * 0.5)**2 + (ruin.height)**2)))
+		new /obj/effect/landmark/ruin/automatic/clearing(central_turf, ruin, round(sqrt((ruin.width) ** 2 + (ruin.height) ** 2) / 2))
```
Comment (AlexMorgan3817) [inline code/modules/overmap/exoplanets/helpers.dm:119]: But this isn't my feedback formula........

**Review by Firefox13 — APPROVED** [review]

**Review by TheShown911 — APPROVED** [review]

**Pull request merged**